src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -21409,6 +21491,1980 @@ static class StaticType {}
+                // (14,7): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                // catch when (TakeOutParam(out var x4) && x4 > 0)
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(14, 7),
+                // (26,7): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                // catch when (TakeOutParam(out var x7) && x7 > 0)
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(26, 7),
+                // (33,7): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                // catch when (TakeOutParam(out var x8) && x8 > 0)
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(33, 7),
+                // (41,7): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                // catch when (TakeOutParam(out var x9) && x9 > 0)
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(41, 7),
+                // (45,11): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                //     catch when (TakeOutParam(out var x9) && x9 > 0) // 2
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(45, 11),
+                // (75,9): error CS0111: Type '<invalid-global-code>' already defines a member called 'when' with the same parameter types
+                //         when (Dummy(TakeOutParam(out var x15), x15))
+                Diagnostic(ErrorCode.ERR_MemberAlreadyExists, "when").WithArguments("when", "<invalid-global-code>").WithLocation(75, 9)
+                    );
+
+                var tree = compilation.SyntaxTrees.Single();
+                Assert.Empty(GetOutVarDeclarations(tree));
+                AssertNoGlobalStatements(tree);
+            }
+        }
+
+        [Fact]
+        public void GlobalCode_Catch_02()
+        {
+            var source =
+@"
+try
+{
+    throw new System.InvalidOperationException();
+}
+catch (System.Exception e) when (Dummy(TakeOutParam(e, out var x1), x1))
+{
+    System.Console.WriteLine(x1.GetType());
+}
+
+static bool Dummy(object y, object z) 
+{
+    System.Console.WriteLine(z.GetType());
+    return true;
+}
+
+static bool TakeOutParam<T>(T y, out T x) 
+{
+    x = y;
+    return true;
+}
+";
+            var compilation = CreateCompilationWithMscorlib45(source, options: TestOptions.ReleaseExe.WithScriptClassName("Script"), parseOptions: TestOptions.Script);
```

Comment:
I didn't understand the value of calling `WithScriptClassName`, given that the default class name is already "Script".